### PR TITLE
[Bug Fix] added param validation in backup properties and some commen…

### PR DESCRIPTION
**Pull request by vishak-ms**
…t fixes

### CODE REVIEW

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs [hunk 1]
```diff
@@ -48,11 +48,11 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                if (!string.IsNullOrEmpty(this.BackupStorageRedundancy))
+                if (this.BackupStorageRedundancy.HasValue)
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: if BackupStorageRedundancy is not set by customer, then this cmdlet exits as success with no-op. Do you want to throw any exception saying storageRedundancy input is must (in future - one of redundancy or dedupe is must)

Comment (vishak-ms) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: added exception handling

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs [hunk 2]
```diff
@@ -48,13 +48,17 @@ public override void ExecuteCmdlet()
...
                 {
                     UpdateVaultStorageTypeRequest vaultStorageRequest = new UpdateVaultStorageTypeRequest();
                     vaultStorageRequest.Properties = new StorageTypeProperties();
-                    vaultStorageRequest.Properties.StorageModelType = BackupStorageRedundancy;
+                    vaultStorageRequest.Properties.StorageModelType = BackupStorageRedundancy.ToString();
                     AzureOperationResponse storageResponse = RecoveryServicesClient.UpdateVaultStorageType(this.Vault.ResouceGroupName, this.Vault.Name, vaultStorageRequest);
                 }
+                else
+                {
+                    this.HandleException(new Exception("You need to provide the BackupStorageRedundancy input parameter"));
```
Comment (swatimadhukar) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: Move this to resource.resx file

Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: yes, pls add in resx file. Also make sure to run all tests and passing before checkin.

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs [hunk 3]
```diff
@@ -48,12 +48,19 @@ public override void ExecuteCmdlet()
...
-                    AzureOperationResponse storageResponse = RecoveryServicesClient.UpdateVaultStorageType(this.Vault.ResouceGroupName, this.Vault.Name, vaultStorageRequest);
+                    vaultStorageRequest.Properties.StorageModelType = BackupStorageRedundancy.ToString();
+                    AzureOperationResponse storageResponse = RecoveryServicesClient.UpdateVaultStorageType(
+                                                                this.Vault.ResouceGroupName, this.Vault.Name, vaultStorageRequest);
+                }
+                else
+                {
+                    throw new ArgumentException(
+                        Properties.Resources.NoBackupPropertiesProvided,
+                        this.BackupStorageRedundancy.ToString());
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs]: null.tostring()

File: src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Properties/Resources.resx [hunk 4]
```diff
@@ -298,4 +298,7 @@ Please provide a storage account with the same location as that of the vault.</v
   <data name="ResourceGroupNameNullOrEmpty" xml:space="preserve">
     <value>Resource Group name mentioned is either null or empty</value>
   </data>
+  <data name="NoBackupPropertiesProvided" xml:space="preserve">
+    <value>You need to provide the BackupStorageRedundancy input parameter</value>
```
Comment (sudhakar4171) [inline src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Properties/Resources.resx]: something like 'backupstorageredundancy is null or empty'

**Pull request merged**